### PR TITLE
Feature/code analysis integration

### PR DESCRIPTION
**Pull request by oren-akto**

### CODE REVIEW

File: apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/observe/api_collections/component/SourceLocation.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        truncatedFilePath = `${topDirectory}/.../${fileName}`
+    }
+
+    const attachLineNo = location.lineNo !== -1 ? `:${location.lineNo}` : ""
+    const filePathText = truncatedFilePath + attachLineNo
+    const tooltipText = location.filePath
+
+    return (
+            <Box width="200px">
+                <Tooltip content={tooltipText} preferredPosition="below">
```
Comment (Ark2307) [inline apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/observe/api_collections/component/SourceLocation.jsx:23]: use TooltipText component here

File: apps/dashboard/src/main/java/com/akto/action/CodeAnalysisAction.java [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+        List<String> trafficApiKeys = new ArrayList<>();
+        for (BasicDBObject trafficApi: trafficApis) {
+            BasicDBObject trafficApiApiInfoKey = (BasicDBObject) trafficApi.get("_id");
+            String trafficApiMethod = trafficApiApiInfoKey.getString("method");
+            String trafficApiUrl = trafficApiApiInfoKey.getString("url");
+            String trafficApiEndpoint = "";
+
+            // extract path name from url
+            try {
+                URL url = new URL(trafficApiUrl);
```
Comment (avneesh-akto) [inline apps/dashboard/src/main/java/com/akto/action/CodeAnalysisAction.java:79]: Use URI if possible

File: apps/dashboard/src/main/java/com/akto/action/CodeAnalysisAction.java [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+                trafficApiEndpoint = url.getPath(); 
+
+                trafficApiEndpoint = new URI(trafficApiEndpoint).getPath()
+                        .replace("%7B", "{")
+                        .replace("%7D", "}");
+            } catch (Exception e) {
+                loggerMaker.errorAndAddToDb("Error parsing URL: " + trafficApiUrl, LogDb.DASHBOARD);
```
Comment (avneesh-akto) [inline apps/dashboard/src/main/java/com/akto/action/CodeAnalysisAction.java:86]: skip the API if parsing fails

Comment (avneesh-akto) [reply apps/dashboard/src/main/java/com/akto/action/CodeAnalysisAction.java:86]: Also add error message.

File: apps/dashboard/src/main/java/com/akto/action/observe/InventoryAction.java [hunk 4]
```diff
@@ -355,8 +356,42 @@ public String fetchAPICollection() {
         if (unused == null) {
             unused = new HashSet<>();
         }
+        
         response.put("unusedEndpoints", unused);
 
+        // Attach code analysis collection
+        BasicDBObject codeAnalysisCollectionInfo = new BasicDBObject();
+        ApiCollection apiCollection = ApiCollectionsDao.instance.findOne(Filters.eq(Constants.ID, apiCollectionId));
```
Comment (avneesh-akto) [inline apps/dashboard/src/main/java/com/akto/action/observe/InventoryAction.java:364]: use `getMeta`

**Pull request merged**